sdk/lib/opentelemetry/sdk/trace/tracer_factory.rb

```diff
@@ -8,7 +8,7 @@ module OpenTelemetry
   module SDK
     module Trace
       # {TracerFactory} is the SDK implementation of {OpenTelemetry::Trace::TracerFactory}.
-      class TracerFactory
+      class TracerFactory < OpenTelemetry::Trace::TracerFactory
```

Comment:
For the formatters? 😞 

I was kinda hoping to rely on 🦆 typing here. I suspect we may want the formatters to be configurable, at which point we can remove the subclassing, but for now this is 👍

Reply:
@fbogsany yes, I wanted to be able to use the `http_text_format` method. I think it would be nice to be able to do:

```ruby
# name is just an example
OpenTelemetry::Propagators::HTTPTextHeader.inject(span.context, request.headers)
```
versus the wiring I need to do now:
```ruby
formatter = OpenTelemetry.tracer_factory.http_text_format
# or
# formatted = OpenTelemetry::DistributedContext::Propagation::HTTPTextFormat.new

formatter.inject(span.context, request.headers) do |carrier, key, value|
  carrier[key] = value
end
```
I know #140 (default getter/setter) will help with this but I was wondering if maybe moving that to a `Propagator` would allow the formatter to stay simple. (cc: @mwear)